zeeguu/api/utils/route_wrappers.py

```diff
@@ -19,16 +24,37 @@ def with_session(view):
     def wrapped_view(*args, **kwargs):
         try:
             session_uuid = flask.request.args["session"]
-            session_object = Session.find(session_uuid)
-            if session_object is None:
+            user_id, timeout = SESSION_CACHE.get(
+                session_uuid,
+                (
+                    None,
+                    datetime.now() - timedelta(0, SESSION_CACHE_EXPIRE),
```

Comment:
is this not the default value? wouldn't it be more intutive to return None, None if we don't have anything in the cache? why do we compute some timeout in the past here? is it important to subtract exact the expiry from here? i might be tired :(

Reply:
It's a good point. The only reason I used a time delta was to not have to check for None in the conditional, so I just set to something that would trigger the update. I now get both None.